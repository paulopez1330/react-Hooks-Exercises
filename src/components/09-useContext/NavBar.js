import React from 'react';
import { Link, NavLink } from "react-router-dom";

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link className="navbar-brand" to="/">useConext</Link>
                
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <NavLink exact activeClassName="active" to="/" className="nav-item nav-link" aria-current="page">Home</NavLink>
                    <NavLink exact activeClassName="active" to="/login" className="nav-item nav-link">Login</NavLink>
                    <NavLink exact activeClassName="active" to="/about" className="nav-item nav-link">About</NavLink>
                </div>
            </div>
        </nav>
    )
}
