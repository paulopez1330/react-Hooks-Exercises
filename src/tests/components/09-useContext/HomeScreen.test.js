import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Prueba de <HomeScreen />', () => {
    
    const user = {
        name: 'paulo lopez',
        email: 'paulo@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            < HomeScreen />
        </UserContext.Provider>
    );
    
    test('debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })
    
})


