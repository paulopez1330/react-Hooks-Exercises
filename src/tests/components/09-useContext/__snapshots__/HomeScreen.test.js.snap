// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <HomeScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "paulo lopez",
   "email": "paulo@gmail.com"
}
  </pre>
</div>
`;
